src/Form/AssignUserForm.php

```diff
@@ -0,0 +1,268 @@
+   */
+  protected $sectionStorage;
+
+  /**
+   * The user section storage service.
+   *
+   * @var \Drupal\workbench_access\UserSectionStorageInterface
+   */
+  protected $userSectionStorage;
+
+  /**
+   * The role section storage service.
+   *
+   * @var \Drupal\workbench_access\RoleSectionStorageInterface
+   */
+  protected $roleSectionStorage;
+
+  /**
+   * Constructs the form object.
+   *
+   * @param \Drupal\workbench_access\WorkbenchAccessManagerInterface $manager
+   *   The workbench access manager.
+   * @param \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $scheme_storage
+   *   The access scheme storage handler.
+   * @param \Drupal\workbench_access\SectionAssociationStorageInterface $section_storage
+   *   The section storage handler.
+   * @param \Drupal\workbench_access\UserSectionStorageInterface $user_section_storage
+   *   The user section storage service.
+   * @param \Drupal\workbench_access\RoleSectionStorageInterface $role_section_storage
+   *   The role section storage service.
+   */
+  public function __construct(WorkbenchAccessManagerInterface $manager, ConfigEntityStorageInterface $scheme_storage, SectionAssociationStorageInterface $section_storage, UserSectionStorageInterface $user_section_storage, RoleSectionStorageInterface $role_section_storage) {
+    $this->manager = $manager;
+    $this->schemeStorage = $scheme_storage;
+    $this->sectionStorage = $section_storage;
+    $this->userSectionStorage = $user_section_storage;
+    $this->roleSectionStorage = $role_section_storage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.workbench_access.scheme'),
+      $container->get('entity.manager')->getStorage('access_scheme'),
+      $container->get('entity.manager')->getStorage('section_association'),
+      $container->get('workbench_access.user_section_storage'),
+      $container->get('workbench_access.role_section_storage')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'workbench_access_assign_user';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, UserInterface $user = NULL) {
+    $account = $this->currentUser();
+    $this->user = $user;
+    $form_enabled = FALSE;
+    $active_schemes = [];
+
+    // Load all schemes.
+    $schemes = $this->schemeStorage->loadMultiple();
+    foreach ($schemes as $scheme) {
+      $user_sections = $this->userSectionStorage->getUserSections($scheme, $user, FALSE);
+      $admin_sections = $this->userSectionStorage->getUserSections($scheme, $account, FALSE);
+      $options = $this->getFormOptions($scheme);
+      $role_sections = $this->roleSectionStorage->getRoleSections($scheme, $user);
+      foreach ($options as $value => $label) {
+        if (in_array($value, $role_sections)) {
```

Comment:
this should use the third argument, as 0 would match any string

Reply:
Done and testing correctly.